pyfolio/sanitize.py

```diff
@@ -0,0 +1,237 @@
+    -----
+    - Must be a pd.DataFrame
+        - If not, raise a ValueError
+    - There must not be any NaNs
+        - If there are, fill NaNs with zeros and warn
+    - Index must be timezone-localized datetimes
+        - If not, localize to UTC and warn
+    - There must a column called 'cash'
+        - If not, raise ValueError
+    - Positions must be floats
+        - if not, coerce values to float and warn
+        - Failing that, raise a ValueError
+    """
+    if not isinstance(positions, pd.DataFrame):
+        msg = '`positions` is not a pd.DataFrame.'
+        raise ValueError(msg)
+
+    if positions.isnull().any().any():
+        positions = positions.fillna(0)
+        msg = '`positions` contains NaNs. Filling with 0...'
+        warn(msg)
+
+    if positions.index.tz is None:
+        positions.index = positions.index.tz_localize('UTC')
+        msg = ('`positions` index is not timezone-localized. '
+               'Localizing to UTC...')
+        warn(msg)
+
+    if 'cash' not in positions.columns:
+        msg = '`positions does not contain a `cash` column.'
+        raise ValueError(msg)
+
+    if not all(positions.apply(lambda col: col.dtype,
+                               axis='columns') == float):
+        try:
+            positions = positions.astype(float)
+            msg = '`positions` does not have float dtype. Coercing to float...'
+            warn(msg)
+        except ValueError:
+            msg = ('`positions` does not have float dtype, and could not be '
+                   'coerced into floats.')
+            raise ValueError(msg)
+
+    return positions
+
+
+def sanitize_txns(txns):
+    """
+    Sanitize transactions.
+
+    Returns
+    -------
+    txns : pd.DataFrame
+        Executed trade volumes and fill prices.
+        - One row per trade.
+        - Trades on different names that occur at the
+          same time will have identical indicies.
+        - Example:
+            index                  amount   price    symbol
+            2004-01-09 12:18:01    483      324.12   'AAPL'
+            2004-01-09 12:18:01    122      83.10    'MSFT'
+            2004-01-13 14:12:23    -75      340.43   'AAPL'
+
+    Notes
+    -----
+    - Must be a DataFrame
+        - If not, raise ValueError
+    - There must not be any NaNs
+        - If there are, raise a ValueError
+    - Index must be timezone-localized datetimes
+        - If not, localize to UTC and warn
+    - There must be columns called 'amount', 'price', and 'symbol'
+        - If not, raise a ValueError
+    - Amounts must be ints
+    - Prices must be floats
+    - Symbols must be strings or ints
+        - If not, warn
+    """
+    if not isinstance(txns, pd.DataFrame):
+        msg = '`txns` is not a pd.DataFrame.'
+        raise ValueError(msg)
+
+    if txns.isnull().any().any():
+        msg = '`txns` contains NaNs.'
+        raise ValueError(msg)
+
+    if txns.index.tz is None:
+        txns.index = txns.index.tz_localize('UTC')
```

Comment:
@twiecki and I agreed that it would be better to coerce datetimes into UTC, simply because you will almost certainly run into an error if the timezones were not localized to something. We thought it more likely that a user just forgot to localize it to something, and perhaps doesn't even care about the timezone, so long as the tear sheet comes back with a result. In fact, thinking about it now, we should probably check that the timezones of all three DataFrames are the same, as that would raise an error too.

Perhaps you could explain why you'd be unhappy if pyfolio coerced the index?

Reply:
I think what @ssanderson means is that we shouldn't in-place change user-passed objects, not that we shouldn't change them after making a copy.

Reply:
@twiecki is right. I'm fine with us making a copy and then localizing the index with a warning, but the issue here is that if I actually meant to have a tz-naive series (which is often actually a better representation for daily data), then I wouldn't expect that passing my series to pyfolio would change my series in place, so I'd be confused and surprised if something like:

```python
some_func_expecting_naive_dates(my_series)
```

worked, but

```python
create_full_tear_sheet(my_series)
some_func_expecting_naive_dates(my_series)
```

failed.

Reply:
I wonder if we should change pyfolio to expect `tz-naive` series everywhere.

Reply:
That's probably the decision I'd make for representing dates in Zipline if I were starting from scratch. 

The problem with using tz-aware dates for date labels is that it makes them comparable with intraday values, which is usually **not** what you want. Especially once you start working with multiple calendars, you usually want a clear distinction between values that represent "dates" and values that represent specific times within those dates.

We haven't changed our representation in Zipline because it'd be too much of a backwards compatibility headache in Zipline/on Quantopian, and there isn't a compelling enough user-facing reason to break old code. The situation is a bit like str/unicode in Python 2 :/.

Reply:
OK, I guess we should defer to zipline then and stay on tz-aware.